src/commands/test.js

```diff
@@ -90,3 +154,122 @@ function getPlatform(logs) {
+  let versionOpt = opts.get("version")
+  let latestVersion = opts.get("latestVersion")
+  let recompileOpt = opts.get("recompile")
+
+  // Remove binary-install-raw binaries, because docker has permission issues
+  // when building the docker images
+  await filesystem.remove("./node_modules/binary-install-raw/bin")
+
+  // Get current working directory
+  let current_folder = await filesystem.cwd()
+
+  // Build the Dockerfile location. Defaults to ./tests/.docker if
+  // a custom testsFolder is not delcared in the subgraph.yaml
+  let dockerDir = ""
+
+  try {
+    let doc = await yaml.load(filesystem.read('subgraph.yaml', 'utf8'))
+    testsFolder = doc.testsFolder || './tests'
+    dockerDir = testsFolder.endsWith('/') ? testsFolder + '.docker' : testsFolder + '/.docker'
+  } catch (error) {
+    print.error(error)
+    return
+  }
+
+  // Create the Dockerfile
+  try {
+    await filesystem.write(`${dockerDir}/Dockerfile`, dockerfile(versionOpt, latestVersion))
+    print.info('Successfully generated Dockerfile.')
+  } catch (error) {
+    print.info('A problem occurred while generating the Dockerfile. Please attend to the errors below:')
+    print.error(error)
+    return
+  }
+
+  // Run a command to check if matchstick image already exists
+  exec('docker images -q matchstick', (error, stdout, stderr) => {
+    // Collect all(if any) flags and options that have to be passed to the matchstick binary
+    let testArgs = ''
+    if (coverageOpt) testArgs = testArgs + ' -c'
+    if (recompileOpt) testArgs = testArgs + ' -r'
+    if (datasource) testArgs = testArgs + ' ' + datasource
+
+    // Build the `docker run` command options and flags
+    let dockerRunOpts = ['run', '-it', '--rm', '--mount', `type=bind,source=${current_folder},target=/matchstick`]
+
+    if(testArgs !== '') {
+      dockerRunOpts.push('-e')
+      dockerRunOpts.push(`ARGS=${testArgs.trim()}`)
+    }
+
+    dockerRunOpts.push('matchstick')
+
+    // If a matchstick image does not exists, the command returns an empty string,
+    // else it'll return the image ID. Skip `docker build` if an image already exists
+    // If `-v/--version` is specified, delete current image(if any) and rebuild.
+    // Use spawn() and {stdio: 'inherit'} so we can see the logs in real time.
+    if(stdout === '' || versionOpt || forceOpt) {
+      if ((stdout !== '' && versionOpt) || forceOpt) {
+        exec('docker image rm matchstick', (error, stdout, stderr) => {
+          print.info(chalk.bold(`Removing matchstick image\n${stdout}`))
+        })
+      }
+      // Build a docker image. If the process has executed successfully
+      // run a container from that image.
+      spawn(
+        'docker',
+        ['build', '--no-cache', '-f', `${dockerDir}/Dockerfile`, '-t', 'matchstick', '.'],
+        { stdio: 'inherit' }
+      ).on('close', code => {
+        if (code === 0) {
+           spawn('docker', dockerRunOpts, { stdio: 'inherit' })
+        }
```

Comment:
You don't need to handle the error here because the stdio is inherited right? Also the option here:

```javascript
{ stdio: 'inherit' }
```

Doesn't need to include `stderr` as well? 🤔

Reply:
According to the docs:
`'inherit': equivalent to ['inherit', 'inherit', 'inherit']`
`'inherit': Pass through the corresponding stdio stream to/from the parent process. In the first three positions, this is equivalent to process.stdin, process.stdout, and process.stderr, respectively. In any other position, equivalent to 'ignore'.`

Reply:
It will try to execute the next `docker run` even if the build fails, and I want it to execute only if the build is successful.